staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -629,6 +640,18 @@ func (c *Cacher) Watch(ctx context.Context, key string, opts storage.ListOptions
 		c.watcherIdx++
 	}()
 
+	if !addedWatcher {
+		// For safety we may close the watcher, but it wasn't yet started at this point.
+		watcher.Stop()
```

Comment:
if we haven't started the watcher yet, let's not make an unnecessary Stop call

Reply:
done